scripts/o3de/tests/test_manifest.py

```diff
@@ -416,7 +420,8 @@ def is_file(path : pathlib.Path) -> bool:
                 patch('pathlib.Path.resolve', self.resolve) as _5, \
                 patch('pathlib.Path.samefile', self.samefile) as _6, \
                 patch('pathlib.Path.open', return_value=io.StringIO(TEST_PROJECT_TEMPLATE_JSON_PAYLOAD)) as _7, \
-                patch('pathlib.Path.is_file', new=is_file) as _8:
+                patch('pathlib.Path.is_file', new=is_file) as _8,\
+                patch('o3de.manifest.get_this_engine_path', side_effect=self.get_this_engine_path) as _9: 
```

Comment:
Why use new for `pathlib.Path.is_file`, but side_effects for `o3de_manifest.get_this_engine_path`?

Reply:
Ah good point, this was copied and I didn't update it

Reply:
I'm going to stick with the staticmethod pattern for this one like the other pathlib mocks because I think we'll may re-use this one for other tests.